File: FrontEnd/src/app/entities/capture.ts
import { Filter } from "./filter";
import { Packet } from "./packet";

export class Capture {
  id: number;
  filter: Filter;
  startDate : Date;
  endDate : Date;
  data : Packet[];
  interfaceId : number;
  
  constructor() {
    this.id = 0;
    this.filter = new Filter();
    this.startDate = new Date();
    this.endDate = new Date();
    this.data = [];
    this.interfaceId = -1;
  }
}

File: BackEnd/app/models/Capture.py
from pydantic import BaseModel
from typing import List
from datetime import date
from models.Filter import Filter
from models.Packet import Packet

class Capture(BaseModel):
    id: int
    filter: Filter
    start_date: date
    end_date: date
    data: List[Packet]
    interfaceId: str
    
